lib/mayaUsd/ufe/UsdLight.h

```diff
@@ -0,0 +1,161 @@
+//
+// Copyright 2022 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#pragma once
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/ufe/UsdSceneItem.h>
+
+#include <pxr/base/tf/token.h>
+#include <pxr/usd/usd/prim.h>
+
+#include <ufe/light.h>
+#include <ufe/path.h>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+//! \brief Interface to control lights through USD.
+class MAYAUSD_CORE_PUBLIC UsdLight : public Ufe::Light
+{
+public:
+    typedef std::shared_ptr<UsdLight> Ptr;
+
+    UsdLight();
+    UsdLight(const UsdSceneItem::Ptr& item);
+    ~UsdLight() override = default;
+
+    // Delete the copy/move constructors assignment operators.
+    UsdLight(const UsdLight&) = delete;
+    UsdLight& operator=(const UsdLight&) = delete;
+    UsdLight(UsdLight&&) = delete;
+    UsdLight& operator=(UsdLight&&) = delete;
+
+    //! Create a UsdLight.
+    static UsdLight::Ptr create(const UsdSceneItem::Ptr& item);
+
+    inline PXR_NS::UsdPrim prim() const
+    {
+        PXR_NAMESPACE_USING_DIRECTIVE
+        TF_AXIOM(fItem != nullptr);
+        return fItem->prim();
+    }
+
+    // Ufe::Light overrides
+    const Ufe::Path&    path() const override;
+    Ufe::SceneItem::Ptr sceneItem() const override;
+    Type                type() const override;
```

Comment:
Something complicated possibly missing from the interface. Do we want to be able to change light types through Ufe?

If so does that mean removing any values that don't apply to the new type? In an undo-able way?

Maybe not being able to change the light type is a reasonable limitation for now, but not having the interface means adding that functionality is an ABI break.

Reply:
I don't see a usecase for that. You cant change type without recreating the object in USD, and you cannot do it in Maya as well.